plugins/modules/system/modprobe.py

```diff
@@ -54,7 +54,84 @@
 import traceback
 
 from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils._text import to_native
+from ansible.module_utils.common.text.converters import to_native
+
+
+class Modprobe(object):
+    def __init__(self, module):
+        self.module = module
+        self.modprobe_bin = module.get_bin_path('modprobe', True)
+
+        self.check_mode = module.check_mode
+        self.desired_state = module.params['state']
+        self.name = module.params['name']
+        self.params = module.params['params']
+
+        self.changed = False
+
+    def load_module(self):
+        command = [self.modprobe_bin]
+        if self.check_mode:
+            command.append('-n')
+        command.extend([self.name] + shlex.split(self.params))
+
+        rc, out, err = self.module.run_command(command)
+
+        if rc != 0:
+            return self.module.fail_json(msg=err, rc=rc, stdout=out, stderr=err, **self.result)
+
+        if self.check_mode or self.module_loaded():
```

Comment:
Per lines 149-150 in the PR, `load_module()` will ***only*** be called when `module_loaded()` is `False`, hence checking this here again is redundant (or the other way around - you could check here, but then remove that check from `main()`).

Reply:
The result of `load_module()` is not static and it's use here is the purpose of this PR. The creator of the original issue observed that builtin modules were always reported as absent by `community.general.modprobe` and the actual `modprobe` command had no effect on the module state hence the task was always reported as `changed` when no change was made. The root cause is that `debian` downstreams have more complexity when checking for builtin modules and the steps performed in `load_module()` are not sufficient. So the true fix would be to enhance those checks, but a much simpler analogue is to check if the module state has changed after loading. Note that for non-builtin modules any errors will result in a non-zero exit code from modprobe and are handled by the `rc` check.

Reply:
Well, after looking at it some more the `modules.builtin` file check really should work at least for `i2c-dev` on debian downstreams. Turns out that just adding `rstrip()` to the line comparisons did the trick so all this bugfix really required was a nine character change, but the module is a little more robust now at least.